app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt

```diff
@@ -0,0 +1,30 @@
+package com.wire.android.di
+
+import android.content.Context
+import com.wire.android.util.DeviceLabel
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
+import com.wire.kalium.logic.CoreLogic
+import com.wire.kalium.logic.feature.auth.LoginUseCase
+import javax.inject.Singleton
+
+
+@Module
+@InstallIn(SingletonComponent::class)
+class CoreLogicModule {
+
+    @Singleton
+    @Provides
+    fun coreLogicProvider(@ApplicationContext context: Context): CoreLogic {
+        val proteusPath = context.getDir("proteus", Context.MODE_PRIVATE).path
+        val deviceLabel = DeviceLabel.label
+        return CoreLogic(applicationContext = context, rootProteusDirectoryPath = proteusPath, clientLabel = deviceLabel)
+    }
+
+    @Singleton
```

Comment:
There's no need to have it as a `@Singleton`, right? 
Maybe for this UseCase the `@Singleton` doesn't hurt. But for others, it may.

Especially if we end up introducing something like I mentioned in another comment, to get the current/active session in order to create the UseCase. If it is Singleton, when we change sessions, it will keep the old UseCase, of the past active session.

In any case, I guess UseCases can come out resource hungry in the end, as they consume Repositories that will be merging databases, networking, caches, etc.
So if we can free these resources when they are no longer being used, it would be for the best.
Also, better to have all UseCases behaving the same in terms of lifecycle, than have some Singletons, and some not.

So, my recommendation is to change to `@Factory`. Not for now, not specifically for this UseCase, but to set an example for the future.